dotnet/src/Extensions/TemplateEngine.Basic/BasicPromptTemplateFactory.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace Microsoft.SemanticKernel.TemplateEngine.Basic;
+
+/// <summary>
+/// Default implementation of <see cref="IPromptTemplateFactory"/> for the semantic-kernel prompt template format.
+/// </summary>
+public sealed class BasicPromptTemplateFactory : IPromptTemplateFactory
```

Comment:
Is there a name of the SK prompt template format? Seems like this is _the_ factory for that format, so it should be named after that format, rather than "Basic".

Reply:
This factory will support the semantic-kernel and f-string formats. I can change the name to `DefaultPromptTemplateFactory` or something else later. We need to agree on what the naming strategy will be as it impacts the reliability package also.

Tracking issue: https://github.com/microsoft/semantic-kernel/issues/3386

Reply:
@markwallace-microsoft might work keeping the single responsibility principle, and keep one "factory" for each format, and having the DefaultPromptTemplateFactory as an specialization of the Aggregate.

Reply:
@RogerBarreto Will do, the next template factory will be Handlebars which will be a seperate package as it introduces some new dependencies. After that I'll work on f-string and will follow the pattern you've suggested.